domain/application/service/application.go

```diff
@@ -584,7 +584,7 @@ func (s *ApplicationService) deleteUnit(ctx domain.AtomicContext, unitName strin
 		cleanup, err := s.secretService.InternalDeleteSecret(ctx, uri, secretservice.DeleteSecretParams{
 			Accessor: secretservice.SecretAccessor{
 				Kind: secretservice.UnitAccessor,
-				ID:   unitName,
+				ID:   unitName.String(),
```

Comment:
**question:** Why don't make the ID a unitName ?

Reply:
The ID here is generic, it is not always a unit name

Reply:
This is probably a sign of poor design of SecretAccessor. We should probably have a different type for each `Kind`. But alas

Reply:
I think you push the comment button too early, so I don't know if you blame alas(tair), or complain in a general form, but alas(...), or if there is missing peace of information (but alas their is something very annoying... )

Anyway it is fine to me :)